pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -162,24 +179,15 @@ def evaluation_step(self, batch: Any, batch_idx: int, dataloader_idx: int) -> Op
         # configure step_kwargs
         step_kwargs = self._build_kwargs(batch, batch_idx, dataloader_idx)
 
-        model_ref = self.trainer.lightning_module
-        model_ref._results = Result()
-
         if self.trainer.testing:
-            model_ref._current_fx_name = "test_step"
+            self.trainer.lightning_module._current_fx_name = "test_step"
             with self.trainer.profiler.profile("test_step"):
                 output = self.trainer.accelerator.test_step(step_kwargs)
         else:
-            model_ref._current_fx_name = "validation_step"
+            self.trainer.lightning_module._current_fx_name = "validation_step"
```

Comment:
Might be cleaner to turn `_current_fx_name ` into a context manager.

Reply:
Let's not do it here, could be done in a future PR